### PR TITLE
Emails to offices

### PR DESCRIPTION
**Pull request by rshorey**
@mileswwatkins I guess perhaps you ought to take at least a quick look at this one given the extent of the changes

### CODE REVIEW

**Comment by mileswwatkins** [comment]
Oh wow, sweet job; reviewing now :eggplant:

File: openstates/co/legislators.py [hunk 1]
```diff
@@ -176,12 +174,13 @@ def scrape(self, chamber, session):
...
-            if "email" in metainf:
-                p['email'] = metainf['email']
-            if "number" in metainf:
-                p.add_office('capitol', 'Capitol Office',
-                             phone=metainf['number'],
-                             address='200 E. Colfax\nDenver, CO 80203'
+
+            phone = metainf['number'] if 'number' in metainf else None
+            email = metainf['email'] if 'email' in metainf else None
+            p.add_office('capitol', 'Capitol Office',
```
Comment (mileswwatkins) [inline openstates/co/legislators.py]: Do we have any way to know what their room number is?

File: openstates/il/legislators.py [hunk 2]
```diff
@@ -67,10 +67,13 @@ def scrape(self, chamber, term):
             # email
             email = leg_doc.xpath('//b[text()="Email: "]')
             if email:
-                leg['email'] = email[0].tail
+                email = email[0].tail.strip()
+            else:
+                email = None
+            print email
```
Comment (mileswwatkins) [inline openstates/il/legislators.py]: Remove debug print.

File: openstates/co/legislators.py (hunk 1 above)
Comment (rshorey) [inline openstates/co/legislators.py]: nope, they just have the main address with no room/box on their legislator pages

File: openstates/il/legislators.py (hunk 2 above)
Comment (rshorey) [inline openstates/il/legislators.py]: fixed

File: openstates/mt/legislators.py [hunk 3]
```diff
@@ -112,6 +118,7 @@ def scrape(self, chamber, term):
 
             office['phone'] = deets.get('phone')
             office['fax'] = deets.get('fax')
+            office['email'] = deets.get('email')
             legislator.add_office(**office)
```
Comment (mileswwatkins) [inline openstates/mt/legislators.py]: Is there any case in which they'll have phone/fax/email in `entry` but not in `deets`?

File: openstates/nj/legislators.py [hunk 4]
```diff
@@ -53,10 +53,9 @@ def scrape(self, term, chambers):
             title = rec["Title"]
             legal_position = rec["LegPos"]
             phone = rec["Phone"] or None
-            if 'Email' in rec:
+            email = None
+            if 'Email' in rec and rec["Email"] != "":
```
Comment (mileswwatkins) [inline openstates/nj/legislators.py]: Couldn't we just do `if rec["Email"]:`? It'll fail if `"Email" not in rec`.

File: openstates/ny/legislators.py [hunk 5]
```diff
@@ -159,9 +163,12 @@ def _split_list_on_tag(lis, tag):
                 email = email.xpath(".//a[contains(@href, 'mailto')]")
                 if email != []:
                     email = email[0]
+                    email = email.text_content().strip()
                 else:
                     email = None
 
+
```
Comment (mileswwatkins) [inline openstates/ny/legislators.py]: The final frontier.

Comment (mileswwatkins) [inline openstates/ny/legislators.py]: ;P

File: openstates/nj/legislators.py (hunk 4 above)
Comment (rshorey) [inline openstates/nj/legislators.py]: yep, fixed. forgot this was a row with consistent headers!

File: openstates/ri/legislators.py [hunk 6]
```diff
@@ -127,7 +128,7 @@ def scrape(self, chamber, term):
                              translate[d['party']],
                              **kwargs)
 
-            leg.add_office('district', 'Address', address=d['address'], phone=phone)
+            leg.add_office('district', 'Dictrict Office', address=d['address'], phone=phone, email=email)
```
Comment (mileswwatkins) [inline openstates/ri/legislators.py]: Do they have no capitol address or government email?

**Comment by mileswwatkins** [comment]
Alright, finished looking it over.

File: openstates/ri/legislators.py (hunk 6 above)
Comment (rshorey) [inline openstates/ri/legislators.py]: does not appear that way

**Comment by rshorey** [comment]
ok i think we're good to go!

File: openstates/mt/legislators.py (hunk 3 above)
Comment (rshorey) [inline openstates/mt/legislators.py]: nope, entry comes from a spreadsheet row and there's no column for email, so I'm going to merge!

**Pull request merged**